nx-dev/ui-blog/src/lib/episode-player.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  AmazonMusicIcon,
+  ApplePodcastsIcon,
+  IHeartRadioIcon,
+  SpotifyIcon,
+} from '@nx/nx-dev/ui-icons';
+import Link from 'next/link';
+
+export function EpisodePlayer({
+  podcastYoutubeId,
+  podcastSpotifyId,
+  amazonUrl,
+  appleUrl,
+  iHeartUrl,
+}: {
+  podcastYoutubeId: string;
+  podcastSpotifyId: string;
+  amazonUrl?: string;
+  appleUrl?: string;
+  iHeartUrl?: string;
+}) {
+  const [viewType, setViewType] = useState<ViewMode>('audio');
+  return (
+    <div className="flex basis-2/3 flex-col items-center justify-center gap-2">
+      {viewType === 'audio' ? (
+        <>
+          <iframe
+            style={{ borderRadius: '12px' }}
+            src={`https://open.spotify.com/embed/episode/${podcastSpotifyId}?utm_source=generator&theme=0`}
+            width="100%"
+            height="152"
+            frameBorder="0"
+            allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
+            loading="lazy"
+            key="audio"
+          ></iframe>
+          <div className="mb-4 mt-8">
+            <PlatformLinks
+              amazonUrl={amazonUrl}
+              appleUrl={appleUrl}
+              iHeartUrl={iHeartUrl}
+              podcastSpotifyId={podcastSpotifyId}
+            />
+          </div>
+        </>
+      ) : (
+        <iframe
+          style={{ borderRadius: '12px' }}
+          width="100%"
+          height="400"
+          src={`https://www.youtube.com/embed/${podcastYoutubeId}?si=8rkgAzJfLfd-hxAA`}
+          title="YouTube video player"
+          frameBorder="0"
+          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+          referrerPolicy="strict-origin-when-cross-origin"
+          allowFullScreen
+          key="video"
+        ></iframe>
+      )}
+      <div className="flex flex-wrap gap-6 sm:gap-4">
+        <button
+          className="space-x-4 whitespace-nowrap rounded-md border border-slate-300 bg-white px-4 py-2 text-lg font-medium text-slate-700 shadow-sm transition focus:ring-offset-2 group-hover:bg-slate-50 group-focus:ring-2 group-focus:ring-blue-500 dark:border-slate-700 dark:bg-slate-800 dark:text-slate-200 dark:group-hover:bg-slate-700 dark:group-focus:ring-sky-500"
+          onClick={() => setViewType(getOpposite(viewType))}
+        >
+          Switch to {getOpposite(viewType) === 'audio' ? 'Audio' : 'Video'}
+        </button>
+        <Link
+          className="space-x-4 whitespace-nowrap rounded-md border border-slate-300 bg-white px-4 py-2 text-lg font-medium text-slate-700 shadow-sm transition focus:ring-offset-2 group-hover:bg-slate-50 group-focus:ring-2 group-focus:ring-blue-500 dark:border-slate-700 dark:bg-slate-800 dark:text-slate-200 dark:group-hover:bg-slate-700 dark:group-focus:ring-sky-500"
+          href="/podcast"
+        >
+          More Podcasts
+        </Link>
+      </div>
+    </div>
+  );
+}
+
+export function PlatformLinks({
```

Comment:
can we maybe just reuse the `ListenOn` (`nx-dev/ui-podcast/src/lib/listen-on.tsx`) component Nicholas created for the `/podcast` page and pass in optional props for customizing the links and adjusting the size?

Reply:
I tried working with this, but hitting circular dependencies between ui-blog and ui-podcast when I mess with this. Can we continue on with this as a separate component for now, and look to adjust when we change the episode player to a markdoc component? I expect that will help here!